packages/server/graphql/mutations/deleteUser.ts

```diff
@@ -64,6 +64,8 @@ export default {
       orgIds.map((orgId) => removeFromOrg(userIdToDelete, orgId, undefined, dataLoader))
     )
     const validReason = reason?.trim().slice(0, 2000) || 'No reason provided'
+    const cachedUser = await dataLoader.get('users').load(user.id)
```

Comment:
how does this work?
`segmentIo.track` doesn't use a dataloader, so I'm not sure how changing this would affect the call.

let's just avoid caches by:
1. set `isRemoved,reasonRemoved` in the DBs immediately
2. after a 30-second timeout, set `email: getDeletedEmail(userId)`